src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -14,6 +14,13 @@ internal sealed class ConnectionPool
 		{
 			cancellationToken.ThrowIfCancellationRequested();
 
+			// if all sessions are used, see if any have been leaked and can be recovered
+			if (m_sessionSemaphore.CurrentCount == 0)
+			{
+				GC.Collect();
+				RecoverLeakedSessions();
```

Comment:
The .NET GC does not invoke objects' `Dispose' methods; that has to be done manually by the consumer (typically using a `using` block).

The .NET finalizer will run a finalizable object's destructor (i.e., `~MyObject()`) after the object has been collected. In the typical implementation of the dispose pattern, the finalizer will call `Dispose(false)`. (So in that sense "the GC calls dispose", but in general `Dispose` is not called deterministically, unlike say C++ destructors.)

I enumerated some reasons why I don't think using a finalizer is the right solution in #251.